functions/src/notifications/publishNotifications.ts

```diff
@@ -88,62 +87,54 @@ export const publishNotifications = functions.firestore
     console.log(`topic type: ${topic.type}`)
 
     const handleNotifications = async (
-      topic: BillNotification | OrgNotification
+      topic: BillHistoryUpdateNotification | TestimonySubmissionNotification
     ) => {
       const notificationFields = createNotificationFields(topic)
 
-      console.log(JSON.stringify(notificationFields))
-
       const topicNameSnapshot = await db
         .collectionGroup("activeTopicSubscriptions")
-        .where("topicName", "==", notificationFields.topicName)
+        .where(
+          "topicName",
+          "in",
+          topic.type === "bill"
+            ? [`bill-${topic.billCourt}-${topic.billId}`]
+            : [
+                `testimony-${topic.orgId}`,
+                `bill-${topic.billCourt}-${topic.billId}`
+              ]
+        )
         .get()
 
-      // Send a testimony notification to all users subscribed to the Bill
-      let billSnapshot
-      if (notificationFields.notification.type !== "bill") {
-        billSnapshot = await db
-          .collectionGroup("activeTopicSubscriptions")
-          .where(
-            "topicName",
-            "==",
-            `bill-${notificationFields.notification.court}-${notificationFields.notification.id}`
-          )
-          .get()
-      }
-
-      const uniqueDocs = new Map()
-
-      // Add documents from topicNameSnapshot to the Map
-      topicNameSnapshot.docs.forEach(doc => {
-        uniqueDocs.set(doc.data().uid, doc.data())
+      const users: { [uid: string]: any } = {}
+
+      // Iterate through the topicNameSnapshots and set the notifications
+      topicNameSnapshot.forEach(subscription => {
+        const { uid, type } = subscription.data()
+        if (topic.type === "testimony" && topic.orgId !== uid) {
+          // If the user does not exist, create new notificationFields for the user
+          if (!users[uid]) {
+            users[uid] = cloneDeep(notificationFields)
+            users[uid].uid = uid
+          }
+
+          // Set isUserMatch or isBillMatch based on the type of notification
+          users[uid].notification[
+            type === "testimony" ? "isUserMatch" : "isBillMatch"
```

Comment:
Is this right? It looks like we set `type = "testimony"` for all TestimonySubmissionNotifications, and we only reach this check for TestimonySubmissionNotifications - so would this ever return "isBillMatch"?

If so, we may need to expand the TopicSubscription's `type` so we can more readily distinguish between the three types of notifications:
* Bill History Update
* Testimony on a Followed Bill
* Testimony by a Followed User

Reply:
So that's what I initially assumed, but the snapshot that we're getting from here isn't the TestimonySubmissionNotification, but actually a TopicSubmission, which is the fields generated when you subscribe to a topic. So the topic determines what type of match the notification is.